src/client/src/app/site/[site]/details/edit-details/edit-details-form.tsx

```diff
@@ -60,43 +61,70 @@ const EditDetailsForm = ({
 
   return (
     <form onSubmit={handleSubmit(submitForm)}>
-      <FormGroup error={errors.name?.message}>
+      <FormGroup
+        error={errors.name ? 'You have not entered a name' : undefined}
+      >
         <TextInput
           id="name"
           label="Site name"
-          {...register('name')}
+          {...register('name', {
+            required: true,
+          })}
         ></TextInput>
       </FormGroup>
 
-      <FormGroup error={errors.address?.message}>
+      <FormGroup
+        error={errors.address ? 'You have not entered an address' : undefined}
+      >
         <TextArea
           id="address"
           label="Site address"
-          {...register('address')}
+          {...register('address', {
+            required: true,
+          })}
         ></TextArea>
       </FormGroup>
 
-      <FormGroup error={errors.latitude?.message || errors.longitude?.message}>
+      <FormGroup
+        error={
+          errors.latitude || errors.longitude
+            ? 'You have not entered valid coordinates'
+            : undefined
+        }
+      >
         <TextInput
           id="latitude"
           label="Latitude"
-          {...register('latitude')}
+          {...register('latitude', {
+            required: true,
+            pattern: DECIMAL_REGEX,
+          })}
         ></TextInput>
         <TextInput
           id="longitude"
           label="Longitude"
-          {...register('longitude')}
+          {...register('longitude', {
+            required: true,
+            pattern: DECIMAL_REGEX,
+          })}
         ></TextInput>
       </FormGroup>
 
-      <FormGroup error={errors.phoneNumber?.message}>
+      <FormGroup
+        error={
+          errors.phoneNumber
+            ? 'You have not entered a valid phone number'
+            : undefined
+        }
+      >
         <TextInput
           id="phoneNumber"
           type="tel"
           label="Phone number"
-          title="Please enter numbers and spaces only."
-          pattern="[0-9 ]*"
-          {...register('phoneNumber')}
+          {...register('phoneNumber', {
+            required: true,
+            pattern: PHONE_NUMBER_REGEX,
+          })}
```

Comment:
Likewise here, rather than pattern validation (which personally I'd prefer but nevermind), we'll have bugs raised against this if the users are allowed to type non-numeric characters